docs/changelog/111709.yaml

```diff
@@ -0,0 +1,5 @@
+pr: 111709
+summary: Adding `RankDocsRetrieverBuilder` and `RankDocsQuery`
+area: Ranking
+type: enhancement
+issues: []
```

Comment:
This almost feels like a non-issue as users consuming the API will never use a "rank docs" query or retriever.